pkg/winsvc/service_windows.go

```diff
@@ -0,0 +1,37 @@
+//go:build windows
+// +build windows
+
+package winsvc
+
+import (
+	wsvc "golang.org/x/sys/windows/svc"
+)
+
+type serviceWindows struct{}
+
+func init() {
+	isService, err := wsvc.IsWindowsService()
+	if err != nil {
+		panic(err)
+	}
+	if !isService {
+		return
+	}
+	go wsvc.Run("", serviceWindows{})
```

Comment:
Why the name is empty here?

Reply:
the name passed to svc.Run is almost irrelevant, because the real service name is determined at service install time, so this would be used as a fallback, if used at all.